pandas/core/arrays/datetimelike.py

```diff
@@ -437,6 +440,20 @@ class DatetimeLikeArrayMixin(ExtensionOpsMixin, AttributesMixin, ExtensionArray)
         _generate_range
     """
 
+    # ------------------------------------------------------------------
+    # NDArrayBackedExtensionArray compat
+
+    @property
+    def _ndarray(self) -> np.ndarray:
+        # NB: A bunch of Interval tests fail if we use ._data
+        return self.asi8
+
+    def _from_backing_data(self, arr: np.ndarray):
+        # Note: we do not retain `freq`
+        return type(self)(arr, dtype=self.dtype)  # type: ignore
```

Comment:
unlike Frame and Series, EAs do not have the concept of _constructor. could this be useful?

the exception is Categorical which uses _constructor. so maybe either extend to all EAs or can we remove from Categorical for consistency.

Reply:
IIUC Categorical has _constructor because it subclasses PandasObject.

`_from_backing_data` is specifically about round-tripping or commutativity; not really a good fit for _constructor.